### PR TITLE
Validate place and dataset ids for submitted entry data

### LINKED ISSUE #625: Place and dataset validation
We should validate place and dataset values on form submission against database values.

Currently they are [validated only for not empty values](https://github.com/okfn/opendatacensus/blob/feature/validation/census/controllers/utils.js#L140-L141).

Since proper validation will likely involves DB queries probably validateData has to be converted to async function.

### PR DESCRIPTION
**Pull request by brew**
`place` and `dataset` values were validated only for empty strings. This PR ensures the submitted values also correspond with valid existing ids in the database.

Most of the commits here are linting fluff. The meaningful work is in this commit: 7887a1a

Fixes #625

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -16,8 +16,12 @@ rules:
     - avoid-escape
   semi:
     - error
-    - never
+    - always
```
Comment (pwalsh) [inline .eslintrc]: I presume this is is because it is the only way to manage with the existing code, right?

File: census/controllers/utils.js [hunk 2]
```diff
@@ -143,21 +142,32 @@ var validateQuestion = function(req, question, parentQuestion, validated) {
...
   var mapped = mappedErrors || false;
 
   req.checkBody('place', 'You must select a Place').notEmpty();
   req.checkBody('dataset', 'You must select a Dataset').notEmpty();
 
   validateQuestion(req, 'exists');
 
-  errors = req.validationErrors(mapped);
+  // place and dataset must exist
+  return Promise.join(
```
Comment (pwalsh) [inline census/controllers/utils.js]: @brew nice you used `.join`

File: debug-server.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+'use strict';
+
+var start = require('./census/app').start;
```
Comment (pwalsh) [inline debug-server.js]: @brew extremely small issue, but, as we are implementing ES6 going forward, `const` is better here.

**Comment by pwalsh** [comment]
@brew LGTM

File: .eslintrc (hunk 1 above)
Comment (brew) [inline .eslintrc]: Precisely. Otherwise I'd have to refactor in a way that touches nearly every line the codebase, jamming up `git blame`.

**Pull request merged**